### ISSUE TITLE
http://xively.github.io/channel-viz/ doesn't work properly

### ISSUE THREAD

**Issue by geometrikal** [xively/channel-viz#7]
Gets stuck on "Loading..."

![screen shot 2013-07-04 at 8 27 17 am](https://f.cloud.github.com/assets/1187528/746504/bfded0a2-e42f-11e2-8754-887ad8ee8ccc.png)

**Comment by pcheek** [comment]
Make sure you're using a master key generated at https://xively.com/users/[username]/keys

> More info about master keys:
> Master API Keys are account-level that can access any object.
> Warning Master API Keys with private access permission can list and edit other keys as well as change their permissions (including giving them greater permission).
> You can create device and channel specific keys in the Developer Workbench.

**Comment by geometrikal** [comment]
Yes, this is with a master key.

**Comment by pcheek** [comment]
If you're having this issue when calling 90 days, a month, one week, or even a single day of data it may take a few moments to retrieve and render the data and graphs, respectively. If after a minute you are still stuck on the loading screen could you provide javascript console logs?

**Comment by geometrikal** [comment]
In the safari javascript console I get the error

2013 06 30 09:02:43  GMT Unknown date format

Perhaps there is a 'bad' datapoint in the feed?

**Comment by geometrikal** [comment]
This is the JSON Xively returns when "Vizualize>>" is pressed:

``` JavaScript
{"id":135647615,"title":"Windometer","private":"true","feed":"https://api.xively.com/v2/feeds/135647615.json","status":"frozen","updated":"2013-06-30T09:00:15.320794Z","created":"2013-06-29T03:59:39.603367Z","creator":"https://xively.com/users/geometrikal","version":"1.0.0","datastreams":[{"id":"AccX","current_value":"1","at":"2013-06-30T09:02:43.855775Z","max_value":"40.0","min_value":"1.0"},{"id":"AccY","current_value":"2","at":"2013-06-30T09:02:43.855775Z","max_value":"400.0","min_value":"1.0"},{"id":"AccZ","current_value":"3","at":"2013-06-30T09:02:43.855775Z","max_value":"10.0","min_value":"1.0"},{"id":"MagX","current_value":"4","at":"2013-06-30T09:02:43.855775Z","max_value":"10.0","min_value":"1.0"},{"id":"MagY","current_value":"5","at":"2013-06-30T09:02:43.855775Z","max_value":"10.0","min_value":"1.0"},{"id":"MagZ","current_value":"6","at":"2013-06-30T09:02:43.855775Z","max_value":"11.0","min_value":"1.0"}],"product_id":"cpUlzjU2WmyAyyN-QR71","device_serial":"MD2PQAMHR4JJ"}
```

**Comment by geometrikal** [comment]
Added pull request to fix this

**Comment by pcheek** [comment]
There shouldn't be any bad dates in the response from Xively, but I suppose its possible. Everything you pasted in (JSON) seems fine...can you include the line reference for the unknown date format error? I'd be surprised if its encountering that error upon returning the feed data, as I'd expect a date problem like this to be encountered when rendering the graphs from the datastream's historical data.

**Comment by geometrikal** [comment]
It is the line with parseISO - line 110 in follows.js   I don't think it is a bad data point, just unnecessary parsing of the date string. In my fork I just commented out the date parsing and it worked. :)

A visualisation that works is this one:  https://github.com/sarfata/xively-viewer    They use moment.js to parse the date, perhaps that is a better solution.

**Comment by errordeveloper** [comment]
Fixed in #8
